### ISSUE TITLE
Macro expander in library top-level

### ISSUE THREAD

**Issue by ktakashi** [ktakashi/sagittarius-scheme#272]
**[Original report](https://bitbucket.org/ktakashi/sagittarius-scheme/issues/267) by Marc Nieper-Wißkirchen (Bitbucket: [Marc Nieper-Wißkirchen](https://bitbucket.org/Marc Nieper-Wißkirchen), ).**

----------------------------------------

Consider the following library:

```scheme
(define-library (foo)
  (import (scheme base))
  (export)
  (begin
    (define-syntax foo
      (syntax-rules ()
        ((foo)
         (begin
           (define set! 42)))))
          
(foo)))
```

When I import this library, Sagittarius complains that I attempt to modify an immutable variable.

This seems to be a bug of the macro expander. The macro expander should hygienically rename \```set!` `` because it is not part of the transformer's input. The renamed version of \```set!` `` would not clash with the imported identifier \```set!` ``

Thanks,

Marc

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Thank you for the report!

I’ve confirmed the behaviour.

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Checking immutable variable modification after renaming identifiers (Fixes #267)
